### DISCUSSION TITLE
is Live-Preview function not working on mac?

### DISCUSSION THREAD

**Discussion by keith331** [slint-ui/slint#5266] [General]
It work on my windows11 laptop but mac m1.

**Comment by hunger** [comment]
Do you use the VSCode extension or how do you use the live preview?

I do no technical reason for it to not work. I would guess we are just not building ARM binaries.

In the VSCode extension you could try to switch to the inline preview. That uses WASM and should just run everywhere;-)

**Comment by keith331** [comment]
I am using the VSCode extension, but the "Show preview" feature does not appear on both MAC M1 and Raspberry Pi OS. Is there a plan for this feature to be supported in the future?

**Reply by ghost** [reply]
I have no trouble with `Show preview` and `VsCode` on MAC M1. Works for me also on Linux ARM. That's strange you cannot see it.

**Reply by hunger** [reply]
This is strange indeed. The language server is adding in the "Start Preview", so that would suggest the language server is not running at all.

VSCode has some "Output" area at the bottom of the screen. That has a drop-down to select the output you want in its title bar (left side). Is there a entry for "Slint LSP" in that drop down? What is the output when you select that entry?

**Reply by keith331** [reply]
Here is the output:
`Error - 8:36:29 PM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 8:36:29 PM] The Slint LSP server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.
[Error - 8:36:29 PM] Slint LSP client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 8:36:29 PM] Restarting server failed
  Message: Pending response rejected since connection got disposed
  Code: -32097 `

**Reply by hunger** [reply]
That is not good at all and definitely a bug on our side :-(

It could be some dependency issue or something, which would explain why it works for @FloVanGH but fails for you.
